bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4396,7 +4396,8 @@ ReplicaImp::ReplicaImp(bool firstTime,
   std::set<NodeIdType> internalClients;
   uint16_t clientId = config_.getnumReplicas() + config_.getnumRoReplicas();
   for (int i = 0; i < config_.getnumOfClientProxies(); ++i) proxyClients.insert(clientId++);
-  for (int i = 0; i < config_.getnumOfExternalClients(); ++i) externalClients.insert(clientId++);
+  for (int i = 0; i < config_.getnumOfExternalClients() + config_.numOfClientServices; ++i)
+    externalClients.insert(clientId++);
```

Comment:
Why is it required? The idea is that no requests will come from client service-id, only from external client-ids.